sources/vega/commands/v1/transaction.proto

```diff
@@ -8,13 +8,26 @@ import "vega/commands/v1/validator_commands.proto";
 import "vega/commands/v1/oracles.proto";
 
 message InputData {
-  // A random number used to provided uniqueness and prevents against replay
-  // attack.
+  // A number to provide uniqueness to prevent accidental replays and,
+  // in combination with `block_height`, deliberate attacks.
+  // A hash of the transaction uniquely identifies the transaction, thus
+  // the nonce provides uniqueness for otherwise identical transactions. 
```

Comment:
// ensuring that the transaction hash uniquely identifies a specific transaction.